code/game/objects/items/reagent_containers/hypospray.dm

```diff
@@ -166,6 +118,74 @@
 
 	return TRUE
 
+///Always draws reagents on right click
+/obj/item/reagent_containers/hypospray/afterattack_alternate(atom/A, mob/living/user)
+	if(!in_range(A, user) || !user.Adjacent(A)) //So we arent drawing reagent from a container behind a window
+		return FALSE
+	can_draw_reagent(A, user, TRUE)
+
+///If it's possible to draw from something. Will draw_blood() when targetting a carbon, or draw_reagent() when targetting a non-carbon
+///Returns TRUE if we successfully draw from something
+/obj/item/reagent_containers/hypospray/proc/can_draw_reagent(atom/A, mob/living/user, alternate_draw_mode)
+	if(!A.reagents)
+		return FALSE
+	if(!istype(user))
+		return FALSE
+	if(!(inject_mode == HYPOSPRAY_INJECT_MODE_DRAW || alternate_draw_mode)) //If we aren't drawing or using right click, return FALSE.
+		return FALSE
+	if(reagents.holder_full())
+		to_chat(user, span_warning("[src] is full."))
+		inject_mode = HYPOSPRAY_INJECT_MODE_INJECT
+		update_icon() //So we now display as Inject
+		return FALSE
+
+	if(iscarbon(A))
+		draw_blood(A, user)
+		return TRUE
+
+	if(isobj(A)) //if not mob
+		draw_reagent(A, user)
+		return TRUE
+
+///Checks if the carbon has blood, then tries to draw blood from it
+/obj/item/reagent_containers/hypospray/proc/draw_blood(atom/A, mob/living/user)
+	var/amount = min(reagents.maximum_volume - reagents.total_volume, amount_per_transfer_from_this)
+	var/mob/living/carbon/C = A
+	if(C.get_blood_id() && reagents.has_reagent(C.get_blood_id()))
+		to_chat(user, span_warning("There is already a blood sample in [src]."))
+		return
+	if(!C.blood_type)
+		to_chat(user, span_warning("You are unable to locate any blood."))
+		return
+	if(C.blood_volume <= BLOOD_VOLUME_SURVIVE)
+		to_chat(user, span_warning("This body doesn't have enough blood to draw from."))
+		return
+	if(ishuman(C))
+		var/mob/living/carbon/human/H = C
+		if(H.species.species_flags & NO_BLOOD)
+			to_chat(user, span_warning("You are unable to locate any blood."))
+			return
+		else
+			C.take_blood(src,amount)
+	else
+		C.take_blood(src,amount)
+	reagents.handle_reactions()
+	user.visible_message("<span clas='warning'>[user] takes a blood sample from [A].</span>",
+						span_notice("You take a blood sample from [A]."), null, 4)
+	on_reagent_change()
+
+///Checks if a container is drawable, then draw reagents from the container
+/obj/item/reagent_containers/hypospray/proc/draw_reagent(atom/A, mob/living/user)
+	if(!A.reagents.total_volume)
+		to_chat(user, "<span class='warning'>[A] is empty.")
+		return
+	if(!A.is_drawable())
+		to_chat(user, span_warning("You cannot directly remove reagents from this object."))
+		return
```

Comment:
Why are these checks in this proc? Its a bit confusing to have checks spread across the two procs, it should probably all be in the can_draw proc.

I think you may have done it this way due to how you currently have the attack proc configured? By having the inject mode check there per my other comment, you won't have that issue.